src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -63,29 +76,29 @@ public void setSpinSpeed(double newSpeed) throws IllegalStateException {
    * Detects if the intake is retracted.
    */
   public boolean isRetracted() {
-    return solenoid.get() == DoubleSolenoid.Value.kReverse;
+    return topSolenoid.get() == DoubleSolenoid.Value.kReverse;
   }
 
   /**
    * Detects if the intake is extended.
    */
   public boolean isExtended() {
-    return solenoid.get() == DoubleSolenoid.Value.kForward;
+    return topSolenoid.get() == DoubleSolenoid.Value.kForward;
```

Comment:
Is extended only if both are extended

Add getTopExtension() and getBottomExtension() that return the DoubleSolenoid.Value enum type in case these values are desired at a later point.